### DISCUSSION TITLE
Cypress app layout to vertical screen.

### DISCUSSION THREAD

**Discussion by ronaldohoch** [cypress-io/cypress#29873] [Questions and Help]
Hi guys, is there a layout option i missed to config on Cypress app to make it split, for example, test in the top part and the website in the bottom part? 

May it be achieved by developing a plugin?

Thank you

![image](https://github.com/user-attachments/assets/7cf30c04-06a7-4c6a-bab4-4a3033459f63)
(1440x2560)

**Comment by portyu9** [comment]
Hi @ronaldohoch 

There is currently no supported Cypress configuration option for changing the Test Runner from the normal **left/right** layout to a **top/bottom** layout.

The Cypress App owns that runner UI:

- Command Log on the left;
- Application/Component Under Test on the right.

`cy.viewport()` and `viewportWidth` / `viewportHeight` change the size or orientation of the **application under test**, but they do not change how the Cypress runner itself arranges those two panes.

A normal Cypress plugin cannot reliably implement this either. Plugins can extend Cypress through supported APIs such as Node events and custom commands, but there is no supported API for replacing or re-layouting the Cypress App's own UI.

You could theoretically modify the runner with custom/unsupported CSS or patch Cypress internals, but that would depend on private implementation details and could break on any Cypress update, so I would not recommend it for a project setup.

For a portrait monitor, the supported options are therefore mainly:

- adjust the Cypress browser/window size;
- adjust the AUT viewport with `cy.viewport()`;
- resize the existing Command Log/preview panes manually where possible.

The attached screenshot could not be retrieved from its old GitHub attachment URL, so I could not inspect the exact visual arrangement shown there. The answer above is based on the requested top/bottom runner layout described in the post and the current Cypress Open Mode API.

Please mark this answer as accepted if it helped, thank you.
